core/src/test/java/io/questdb/griffin/SqlParserTest.java

```diff
@@ -1393,6 +1393,22 @@ public void testCreateTableNoCache() throws SqlException {
                         "PARTITION by YEAR");
     }
 
+    @Test
+    public void testGroupByConstantMatchingColumnName() throws SqlException {
+        assertQuery(
+                "select-virtual 'nts' nts, min from (select-group-by [min(nts) min] min(nts) min from (select [nts] from tt timestamp (dts) where nts > '2020-01-01T00:00:00.000000Z'))",
+                "select 'nts', min(nts) from tt where nts > '2020-01-01T00:00:00.000000Z'",
+                modelOf("tt").timestamp("dts").col("nts", ColumnType.TIMESTAMP));
+    }
+
+    @Test
+    public void testGroupByConstantFunctionMatchingColumnName() throws SqlException {
+        assertQuery(
+                "select-virtual now() now, min from (select-group-by [min(nts) min] min(nts) min from (select [nts] from tt timestamp (dts) where nts > '2020-01-01T00:00:00.000000Z'))",
```

Comment:
The current optimization will only check for functions that are definitely runtime constant whatever their arguments (if any). The dateadd function does not fit this category, it would require more complex analysis to determine if its arguments will result in a runtime constant.